### PR TITLE
finalizing database schema

### PR DESCRIPTION
**Pull request by helloseyedjafari**
lot's of refactoring done in this part.
The calligraphy table has been refactored as well. 

lots types have been added to the schema to avoid mixing values.
basic setup for creating a database is done as a test. please checkout `InsertQuranTextTest` file.

due to a lot of refactorings, the previous database tests are invalid and has to be re-written

### CODE REVIEW

File: db/src/commonMain/kotlin/com/islamversity/db/model/SurahRevealedType.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    data class MECCAN (
+        val nameInCalligraphy : String
+    ): SurahRevealedType(nameInCalligraphy)
+    data class MEDINAN (
+        val nameInCalligraphy : String
+    ): SurahRevealedType(nameInCalligraphy)
+}
+
+enum class RevealTypeFlag(val raw : String){
```
Comment (rezaiyan) [inline db/src/commonMain/kotlin/com/islamversity/db/model/SurahRevealedType.kt]: `StringDef `isn't better?

File: db-test/src/androidTest/java/com/islamversity/db_test/db_filler/InsertQuranTextTest.kt [hunk 2]
```diff
@@ -36,15 +30,16 @@ class InsertQuranTextTest {
...
-        assertThat(file.exists()).isTrue()
-        driver.close()
-        file.delete()
+//        val file = File(InstrumentationRegistry.getInstrumentation().targetContext.getNoBackupFilesDir(), dbName)
+//        assertThat(file.exists()).isTrue()
+//        driver.close()
+//        file.delete()
     }
 
     @Test
```
Comment (rezaiyan) [inline db-test/src/androidTest/java/com/islamversity/db_test/db_filler/InsertQuranTextTest.kt:45]: There is no assertion

File: db/src/commonMain/kotlin/com/islamversity/db/model/Calligraphy.kt [hunk 3]
```diff
@@ -53,13 +57,17 @@ data class Calligraphy(
          */
         operator fun invoke(code: String): Calligraphy {
             val codes = code.split("_")
-            if (codes.size < 2) {
-                throw IllegalArgumentException("a calligraphy is created using 3 parts: languageCode, levelType, calligraphyCode")
+            if (codes.isEmpty()) {
```
Comment (rezaiyan) [inline db/src/commonMain/kotlin/com/islamversity/db/model/Calligraphy.kt:60]: In the [LanguageCode](url) the `isValid `condition has not been updated

```
     val isValid: Boolean
        get() =
            lang.length == 2 &&
                    lang == lang.toLowerCase()
```

File: db/src/commonMain/kotlin/com/islamversity/db/model/RevealType.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    data class MECCAN (
+        val nameInCalligraphy : String
+    ): RevealType(nameInCalligraphy)
+    data class MEDINAN (
+        val nameInCalligraphy : String
+    ): RevealType(nameInCalligraphy)
+}
+
+enum class RevealTypeFlag(val raw : String){
```
Comment (rezaiyan) [inline db/src/commonMain/kotlin/com/islamversity/db/model/RevealType.kt:24]: `StringDef `isn't better?

File: db/src/commonMain/kotlin/com/islamversity/db/model/SajdahTypeFlag.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    NONE("NONE"),
+    RECOMMENDED("RECOMMENDED"),
+    OBLIGATORY("OBLIGATORY"),
+    ;
+
+    companion object{
+        operator fun invoke(raw : String) : SajdahTypeFlag =
+            fromRaw(raw)
+
+        fun fromRaw(raw : String) : SajdahTypeFlag =
```
Comment (rezaiyan) [inline db/src/commonMain/kotlin/com/islamversity/db/model/SajdahTypeFlag.kt]: `private`

File: db/src/commonMain/kotlin/com/islamversity/db/model/Calligraphy.kt (hunk 3 above)
Comment (helloseyedjafari) [reply db/src/commonMain/kotlin/com/islamversity/db/model/Calligraphy.kt:60]: please consider nullable `CalligraphyName`

File: db-test/src/androidTest/java/com/islamversity/db_test/db_filler/InsertQuranTextTest.kt (hunk 2 above)
Comment (helloseyedjafari) [reply db-test/src/androidTest/java/com/islamversity/db_test/db_filler/InsertQuranTextTest.kt:45]: This is not actually a test!
this is the fastest way I could create an Android compatible database (to ship with the app).
after this function has run
I'll go the database location on the emulator and copy it to our assets! 🤦

**Pull request merged**